xbmc/pictures/PictureFolderImageFileLoader.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PictureFolderImageFileLoader.h"
+
+#include "FileItem.h"
+#include "Picture.h"
+#include "ServiceBroker.h"
+#include "TextureCache.h"
+#include "filesystem/Directory.h"
+#include "guilib/Texture.h"
+#include "settings/Settings.h"
+#include "settings/SettingsComponent.h"
+#include "utils/FileExtensionProvider.h"
+
+using namespace XFILE;
+
+bool CPictureFolderImageFileLoader::CanLoad(std::string specialType) const
```

Comment:
```c++
const std::string& specialType
```

Reply:
Should I do this in both functions? That is the way the whole interface works, do I need to change all implementations in this PR?

Reply:
Yes, you should. We want to avoid calling the std::string copy constructor unnecessarily.

Here's an example where you could pass by value:

https://github.com/xbmc/xbmc/blob/master/xbmc/games/controllers/types/ControllerNode.cpp#L85-L88

```c++
void CControllerNode::SetPortAddress(std::string portAddress)
{
  m_portAddress = std::move(portAddress);
}
```

In that case, we call the copy constructor on the "outside" because we use std::move() to avoid calling it on the inside of the function. You can then call `SetPortAddress(std::move(portAddress))` on the outside and avoid copying across the entire chain.

But because we don't acquire ownership inside `CPictureFolderImageFileLoader::CanLoad()` the copy would be wasted.

Reply:
Thanks. Done.